### PR TITLE
Added dynamic accessor and mutators from $uuidAttributes

### PR DESCRIPTION
**Pull request by westphalen**
This solves the problem in #33. I have tested it in my own project, but haven't written any unit tests yet.

### CODE REVIEW

File: src/HasBinaryUuid.php [hunk 1]
```diff
@@ -73,7 +73,54 @@ public static function decodeUuid(string $binaryUuid): string
 
     public function toArray()
     {
-        return array_merge(parent::toArray(), [$this->getKeyName() => $this->uuid_text]);
+        $uuidAttributes = $this->getUuidAttributes();
+
+        $array = parent::toArray();
+        foreach ($uuidAttributes as $attributeKey) {
```
Comment (brendt) [inline src/HasBinaryUuid.php:79]: Add an newline before this `foreach` statement.

File: src/HasBinaryUuid.php [hunk 2]
```diff
@@ -73,7 +73,54 @@ public static function decodeUuid(string $binaryUuid): string
...
+            }
+        }
+
+        return $array;
+    }
+
+    public function getUuidAttributes()
+    {
+        $uuidAttributes = [$this->getKeyName()];
+        if (property_exists($this, 'uuidAttributes')) {
```
Comment (brendt) [inline src/HasBinaryUuid.php:91]: Add a newline before this `if` statement.

File: src/HasBinaryUuid.php [hunk 3]
```diff
@@ -73,7 +73,54 @@ public static function decodeUuid(string $binaryUuid): string
...
+            $uuidAttributes = $this->uuidAttributes === null ? [] : $this->uuidAttributes;
+        }
+
+        return $uuidAttributes;
+    }
+
+    public function getAttribute($key)
+    {
+        if (($uuidKey = $this->uuidTextAttribute($key)) && $this->{$uuidKey} !== null) {
```
Comment (brendt) [inline src/HasBinaryUuid.php:100]: While these shorthand assignments are useful for easy writing, it makes it a lot more difficult to read the code. I'd prefer an extra assignment above this `if` statement for clarity.

File: src/HasBinaryUuid.php [hunk 4]
```diff
@@ -73,7 +73,54 @@ public static function decodeUuid(string $binaryUuid): string
...
+        if ($uuidKey = $this->uuidTextAttribute($key)) {
+            $value = static::encodeUuid($value);
+        }
+
+        return parent::setAttribute($key, $value);
+    }
+
+    protected function uuidTextAttribute($key)
+    {
+        if (substr($key, -5) == '_text' &&
```
Comment (brendt) [inline src/HasBinaryUuid.php:118]: Re-structure this `if` statement for better readability.

```
$isTextualAttribute = substr($key, -5) == '_text' ;

if (!$isTextualAttribute) {
    return false;
}

$uuidKey = substr($key, 0, -5);

$isUuidAttribute = in_array($uuidKey, $this->getUuidAttributes());

if ($isUuidAttribute) {
    return $uuidKey;
}
```

File: src/HasBinaryUuid.php [hunk 5]
```diff
@@ -73,7 +73,54 @@ public static function decodeUuid(string $binaryUuid): string
 
     public function toArray()
     {
-        return array_merge(parent::toArray(), [$this->getKeyName() => $this->uuid_text]);
+        $uuidAttributes = $this->getUuidAttributes();
+
+        $array = parent::toArray();
+        foreach ($uuidAttributes as $attributeKey) {
+            if (array_key_exists($attributeKey, $array)) {
+                $array[$attributeKey] = $this->{"{$attributeKey}_text"};
```
Comment (brendt) [inline src/HasBinaryUuid.php:81]: It would be nice if we shouldn't have to hard code the `_text` naming..

**Review by brendt — CHANGES_REQUESTED** [review]
Thanks. This seems like a way better solution than what's pretty hard-coded now.

I made a few suggestions regarding coding style, and would also like to avoid the hard-coded `_text` naming; but I can't think of a good solution at the moment. Suggestions?

Also I'd like to see some tests here :)

**Comment by brendt** [comment]
@westphalen Any updates on this? Do you still want to go through with it?

**Comment by rjsworking** [comment]
Hi everyone

This is a nice feature. Go ahead with it pls.

Regards

**Comment by westphalen** [comment]
@brendt I don't have time to look more into this, as I ended up not using the package anyway. You're welcome to use the pull request still.

**Comment by ragingdave** [comment]
I could pick this back up as I have a pretty urgent usage of the majority of this functionality. I'm just not sure how to pick up a PR from someone else.....Does it mean starting a new PR from my own fork and going from there?

**Comment by freekmurze** [comment]
@ragingdave 

> Does it mean starting a new PR from my own fork and going from there?

Yup.

I'll close this PR as there has been no activity in quite some time.

File: src/HasBinaryUuid.php (hunk 5 above)
Comment (isocroft) [reply src/HasBinaryUuid.php:81]: Yes true... i agree with @brendt

**Pull request closed without merging**